### ISSUE TITLE
Issue parsing `DateTime` with "reduced precision"

### ISSUE THREAD

**Issue by jakevossen5** [chronotope/chrono#528]
I see that in the limitations there is a "Chrono inherently does not support an inaccurate or partial date and time representation" statement, but, I am having to deal with 8601 dates of the form "2020-01-20T05Z", which should be at hour five of January 5th 2021.

I realize that there isn't a great way to represent these in Chrono, but I would like to propose a couple solutions.

1. Return a`DateTime` representing the earliest possible moment that it could be, and a `DateTime` representing the latest possible moment (or a `Duration`).
2. Add some parameter to the parse functions (or create new ones) that allow for some rounding (as in, 05 becomes 05:00:00, 05:30:00, or 05:59:59 depending on what you select).

Would this be something that this project would be implement? If I am given some guidance on what direction to go in, I could probably work on this and try and open up a PR.

**Comment by quodlibetor** [comment]
Yes this is very much doable, but a bit ugly, and something I would love!

The main thing that we need to do is to update [`Parsed`](https://github.com/chronotope/chrono/blob/3467172c31188006147585f6ed3727629d642fed/src/format/parsed.rs#L18-L27) to have optional default arguments.

The main issue is that we need to construct a DateTime from a non-overlapping set of fields, [`to_naive_datetime_with_offset`](https://github.com/chronotope/chrono/blob/3467172c31188006147585f6ed3727629d642fed/src/format/parsed.rs#L571) is the most-inclusive method, and it is a bit of a beast.

That said, I think that many of the places where we return errors in the various `Parsed::to_*` methods could check for a default before actually erroring.

If we add some `default_*` fields to `Parsed` then we could expose them through some public APIs and make the whole thing more pleasant.

Any step on this path would be immensely appreciated!

**Comment by jakevossen5** [comment]
Thanks for the guidance! I will take a hack at this, and let you know if I run into any issues.

**Comment by quodlibetor** [comment]
Sounds great!
